src/lib.rs

```diff
@@ -70,10 +70,52 @@ pub static README_PATHS: &[&str; 5] = &[
 
 pub static LICENSE_PATHS: &[&str; 3] = &["LICENSE", "LICENSE.md", "COPYING"];
 
+#[derive(Debug, Default, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PackageName {
+    pub namespace: String,
+    pub name: String,
+}
+
+impl PackageName {
+    pub fn parse(s: &str) -> Result<Self, &'static str> {
```

Comment:
Let's introduce a `ParsePackageNameError` type instead of using bare strings.

Reply:
ok done